engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -834,34 +833,46 @@ public void pushdownFilter(
                     final long locationEndKey = getLastRowKey(entry.regionIndex);
 
                     try (final WritableRowSet locationInput =
-                            input.subSetByKeyRange(locationStartKey, locationEndKey)) {
+                            selectionCopy.subSetByKeyRange(locationStartKey, locationEndKey)) {
                         locationInput.shiftInPlace(-locationStartKey); // Shift to the region's key space
                         locationInput.retain(entry.rowSetAtLastUpdate); // intersect in place with region's row set
+                        final long locationInputSize = locationInput.size();
 
                         final Consumer<PushdownResult> resultConsumer = result -> {
                             try (final PushdownResult ignored = result) {
+                                final WritableRowSet match = result.match();
+                                final WritableRowSet maybeMatch = result.maybeMatch();
                                 // Add the results to the global set.
-                                if (result.match().isNonempty()) {
-                                    synchronized (matchBuilder) {
-                                        result.match().shiftInPlace(locationStartKey);
-                                        matchBuilder.addRowSet(result.match());
-                                    }
+                                // TODO: This is a bit dangerous, but seems "ok" in context?
+                                if (match.isNonempty()) {
+                                    match.shiftInPlace(locationStartKey);
                                 }
-                                if (result.maybeMatch().isNonempty()) {
-                                    synchronized (maybeMatchBuilder) {
-                                        result.maybeMatch().shiftInPlace(locationStartKey);
-                                        maybeMatchBuilder.addRowSet(result.maybeMatch());
-                                        maybeMatchCount.add(result.maybeMatch().size());
-                                    }
+                                if (maybeMatch.isNonempty()) {
+                                    maybeMatch.shiftInPlace(locationStartKey);
                                 }
+                                builder.add(locationIdx, locationInputSize, result);
                             }
                         };
+                        // TODO: fullSet is not shifted here, and is a lie?
+                        // Should it actually be a subset in the parallelized case?
+                        // Or, can we get rid of it?
+                        // fullSet.subSetByKeyRange(locationStartKey, locationEndKey).shiftInPlace(-locationStartKey)?
```

Comment:
I think we should remove `fullSet`. Any filter that uses positions or keys should never be pushed down. This likely involves communicating with @nbauernfeind to make sure we properly ID these filters and avoid pushdown.